Product/Production/Adapters/General/CONNECTAdminGUI/src/main/java/gov/hhs/fha/nhinc/admingui/services/impl/DocumentQueryServiceImpl.java

```diff
@@ -144,4 +148,51 @@ private AdhocQueryRequest createAdhocQueryRequest(DocumentMetadata query) {
         requestBuilder.build();
         return requestBuilder.getMessage();
     }
+    
+    private void setQueryPatientId(DocumentMetadata query, II correlatedId) {
+        if(correlatedId == null || isPassthrough(query.getOrganization())) {
+            requestBuilder.setPatientId(query.getPatientId());
+            requestBuilder.setPatientIdRoot(query.getPatientIdRoot());           
+        } else {
+            requestBuilder.setPatientId(correlatedId.getExtension());
+            requestBuilder.setPatientIdRoot(correlatedId.getRoot());
+        }
+    }
+    
+    private NhincConstants.UDDI_SPEC_VERSION getOrganizationSpecVersion(String hcid) {
+        final NhincConstants.UDDI_SPEC_VERSION DEFAULT_VERSION = NhincConstants.UDDI_SPEC_VERSION.SPEC_3_0;
+        
+        try {
+            OrganizationType org = ExchangeManager.getInstance().getOrganization(hcid);
+            EndpointType ep = ExchangeManagerHelper.findEndpointTypeBy(org, NhincConstants.DOC_QUERY_SERVICE_NAME);
+            if(ep != null) {
+                return ExchangeManagerHelper.getHighestUDDISpecVersion(ExchangeManagerHelper.getSpecVersions(ep));
+            }
+        } catch (ExchangeManagerException ex) {
+           LOG.warn("Unable to find highest spec version for organization {}, setting to {}", hcid, DEFAULT_VERSION, ex);
+        }
+        return DEFAULT_VERSION;
+    } 
+    
+    private boolean isPassthrough(String hcid) {
+        String propertyName;
+        NhincConstants.UDDI_SPEC_VERSION specVersion = getOrganizationSpecVersion(hcid);
+        if(specVersion.equals(NhincConstants.UDDI_SPEC_VERSION.SPEC_2_0)) {
+            propertyName = "docquery.20.outboundDocQuery";
```

Comment:
I feel like it would be beneficial to use constants here for something like this.